go/vt/servenv/grpc_vitess_static_auth_plugin.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servenv
+
+import (
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+
+	log "github.com/golang/glog"
+)
+
+var (
+	credsFile                = flag.String("grpc_vitess_static_auth_plugin_file", "", "JSON File to read the users/passwords from.")
+	requireTransportSecurity = flag.Bool("grpc_vitess_static_auth_require_transport_security", false, "when true it requires transport security. Should be use in conjuction with tls certificates")
+)
+
+type VitessAuthEntry struct {
+	Username string
+	Password string
+	// TODO (@rafael) Add authorization parameters
+}
+
+func (e *VitessAuthEntry) GetRequestMetadata(context.Context, ...string) (map[string]string, error) {
+	return map[string]string{
+		"username": e.Username,
+		"password": e.Password,
+	}, nil
+}
+
+func (c *VitessAuthEntry) RequireTransportSecurity() bool {
+	return *requireTransportSecurity
+}
+
+type VitessStaticAuth struct {
+	entries []VitessAuthEntry
+}
+
+func (srvAuth *VitessStaticAuth) Authenticate(ctx context.Context, fullMethod string) (context.Context, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
```

Comment:
We might want to consider adding something to periodically check and reparse the file if it changes. Otherwise the revocation story is a bit weak.

Reply:
We probably also want a mutex to protect the entries  since I suspect the GRPC can run in different threads.
Better safe than sorry anyway.